web/src/app/modules/shared/components/presentation/select-file/select-file.component.ts

```diff
@@ -47,19 +58,31 @@ export class SelectFileComponent
     this.action = view.config.action;
   }
 
-  inputFileChanged(event) {
-    if (event.target.files && event.target.files[0]) {
+  inputFileChanged(event: Event) {
+    const files = (event.target as HTMLInputElement).files;
+    const fileList: File[] = [];
+    if (files && files[0]) {
       if (this.fileChanged) {
-        this.fileChanged.emit(event.target.files);
+        this.fileChanged.emit(files);
       }
       if (isDevMode()) {
-        console.log('Selected file(s):', event.target.files);
+        console.log('Selected file(s):', files);
+      }
+      for (let i = 0; i < files.length; i++) {
+        const file = files.item(i);
+        fileList.push({
+          name: file.name,
+          type: file.type,
+          path: file.path,
+          lastModified: file.lastModified,
+          size: file.size,
+        });
       }
 
       if (this.action) {
         this.actionService.perform({
           action: this.action,
-          files: event.target.files,
+          files: fileList,
```

Comment:
The intention of the original implementation was to pass the selection data as is, and allow consumer to directly access not only the selected file(s) metadata (defined in `lib.dom.d.ts`), but also the file content thought `Blob` interfaces. This approach makes copy of original file metadata, and removes ability to read file with original `Blob` interfaces. This is considerable loss for typescript consumer - how are they going to read the file with our metadata that does not contain either path info or `Blob` steaming interfaces?

Reply:
The file object isn't serializable. See: https://stackblitz.com/edit/angular-ivy-85wt7e?file=src/app/app.component.ts


Under existing usage, we  use a different code path entirely (https://github.com/vmware-tanzu/octant/blob/master/web/src/app/modules/shared/components/smart/editor/editor.component.ts#L92). Consumers of this action will have to take the filepath  from this action and read it locally . In other words, this component would only see meaningful usage on electron.


iirc, it was decided explicitly not to send the file over the wire.